server/auth.go

```diff
@@ -1418,8 +1419,14 @@ func comparePasswords(serverPassword, clientPassword string) bool {
 		if err := bcrypt.CompareHashAndPassword([]byte(serverPassword), []byte(clientPassword)); err != nil {
 			return false
 		}
-	} else if serverPassword != clientPassword {
-		return false
+	} else {
+		// stringToBytes should be constant-time near enough compared to
+		// turning a string into []byte normally.
+		spass := stringToBytes(serverPassword)
+		cpass := stringToBytes(clientPassword)
+		if subtle.ConstantTimeCompare(spass, cpass) == 0 {
```

Comment:
So this will slow down reconnect yes?

Reply:
Should be marginal, standard string comparison will give up on the first mismatching character, whereas `ConstantTimeCompare` iterates over all characters before returning.

Reply:
Should be about no difference in time for valid credentials. Should only make invalid credential rejection slower.